src/borg/archiver.py

```diff
@@ -1122,7 +1122,16 @@ def do_diff(self, args, repository, manifest, key, archive):
         """Diff contents of two archives"""
 
         def print_json_output(diff, path):
-            print(json.dumps({"path": path, "changes": [j for j, str in diff]}, sort_keys=True, cls=BorgJsonEncoder))
+            def actual_change(j):
+                if j["type"] == "modified":
+                    # It's useful to show 0 added and 0 removed for text output
+                    # but for JSON this is essentially noise. Additionally, the
+                    # JSON key is "changes", and this is not actually a change.
+                    return not (j["added"] == 0 and j["removed"] == 0)
+                else:
+                    return True
```

Comment:
This looks like it could be done with a single boolean expression.

Not sure that text should behave differently than json btw. In the end, it is borg **diff**, so it should show changes, not not-changes.

Reply:
You mean `(j["added"] == j["removed"] == 0)`?

Alright, I'll fix it for text-based output.

Reply:
no, i mean 1126..1132.

Reply:
Guess this would behave the same:
```
return  j["type"] != "modified" or not (j["added"] == 0 and j["removed"] == 0)
```

Reply:
The exact line I came up with locally :P. Still need to fix the tests...

Reply:
> Guess this would behave the same:
> 
> ```
> return  j["type"] != "modified" or not (j["added"] == 0 and j["removed"] == 0)
> ```

This one-liner ended up not being possible; if comparing two archives with different chunking params, the `added` and `removed` keys won't exist. I went with:

```py
if j["type"] != "modified":
    return True
else:
    # Added/removed keys will not exist if chunker params differ
    # between the two archives. Err on the side of caution and assume
    # a real modification in this case (short-circuiting retrieving
    # non-existent keys).
    return not {"added", "removed"} <= j.keys() or not (j["added"] == 0 and j["removed"] == 0)
```

Not clear to me it's worth the one-liner, but if you can massage the logic a bit, happy to make changes.

Reply:
Yeah, keep it like that, gets too long/complicated as a one-liner.